Trenches - Terrain Trenching

PabstMirror opened the pull request:
very rough draft

https://www.youtube.com/watch?v=Qfz-AD6txNc

dig trenches in to the ground
- ONLY straight east/north

![yuck](https://user-images.githubusercontent.com/9376747/217643067-ad1bc249-d6c4-4b3d-a443-5eeaf14d89b6.png)
we drop the terrain on the red dots
then place these terrain blocks in the green to create the edge we want

Drofseh commented:
Do you think it'd be worth adding grass cutters fish the centre of the trench?
There aren't normally plants in the bottom of recently dug trenches lol

654wak654 commented:
> Do you think it'd be worth adding grass cutters to the centre line of the trench? There aren't normally plants in the bottom of recently dug trenches lol

![image](https://media.tenor.com/uJAd6-kG9x4AAAAd/life-uh.gif)

654wak654 commented:
> Do you think it'd be worth adding grass cutters to the centre line of the trench? There aren't normally plants in the bottom of recently dug trenches lol

![image](https://media.tenor.com/uJAd6-kG9x4AAAAd/life-uh.gif)

veteran29 commented:
Grass cutters mess with AI pathing IIRC, but trench objects most likely do so too.

veteran29 commented:
Grass cutters mess with AI pathing IIRC, but trench objects most likely do so too.

PabstMirror commented:
added a third block for the floor, so it's a nice dirt texture and flat bottom
added option to run grass cutters (but I think a little clutter helps theses look less blocky)

![test](https://user-images.githubusercontent.com/9376747/221021255-9d159de9-5414-4201-9501-0afb26be189c.jpg)

jonpas commented on addons/zeus/functions/fnc_moduleLayTrench.sqf:
```diff
@@ -0,0 +1,51 @@
+_logic setPos _startPos;
+
+
+// todo: it would be nice to show visually that trenches can only be dug north/south or east/west
```
Put it in the tooltip of the module?
